go/codepipeline-build-deploy/codepipeline-build-deploy.go

```diff
@@ -0,0 +1,326 @@
+			targetGroupBlue,
+		},
+	})
+
+	// Creates an ECS Fargate service
+	fargateService := ecs.NewFargateService(stack, jsii.String("FargateService"), &ecs.FargateServiceProps{
+		DesiredCount:   jsii.Number(1),
+		ServiceName:    jsii.String("fargate-frontend-service"),
+		TaskDefinition: fargateTaskDef,
+		Cluster: ecs.NewCluster(stack, jsii.String("EcsCluster"), &ecs.ClusterProps{
+			EnableFargateCapacityProviders: jsii.Bool(true),
+			Vpc:                            clusterVpc,
+		}),
+		// Sets CodeDeploy as the deployment controller
+		DeploymentController: &ecs.DeploymentController{
+			Type: ecs.DeploymentControllerType_CODE_DEPLOY,
+		},
+	})
+
+	// Adds the ECS Fargate service to the ALB target group
+	fargateService.AttachToApplicationTargetGroup(targetGroupBlue)
+
+	// Creates new pipeline artifacts
+	sourceArtifact := pipeline.NewArtifact(jsii.String("SourceArtifact"))
+	buildArtifact := pipeline.NewArtifact(jsii.String("BuildArtifact"))
+
+	// Creates an AWS CodePipeline with source, build, and deploy stages
+	pipeline.NewPipeline(stack, jsii.String("BuildPipeline"), &pipeline.PipelineProps{
+		PipelineName: jsii.String("ImageBuildDeployPipeline"),
+		Stages: &[]*pipeline.StageProps{
+			&pipeline.StageProps{
+				StageName: jsii.String("Source"),
+				Actions: &[]pipeline.IAction{
+					pipelineactions.NewCodeCommitSourceAction(&pipelineactions.CodeCommitSourceActionProps{
+						ActionName: jsii.String("CodeCommit"),
+						Branch:     jsii.String("main"),
+						Output:     sourceArtifact,
+						Repository: codeRepo,
+					}),
+				},
+			},
+			&pipeline.StageProps{
+				StageName: jsii.String("Build"),
+				Actions: &[]pipeline.IAction{
+					pipelineactions.NewCodeBuildAction(&pipelineactions.CodeBuildActionProps{
+						ActionName: jsii.String("DockerBuildPush"),
+						Input:      pipeline.NewArtifact(jsii.String("SourceArtifact")),
+						Project:    initialBuild,
+						Outputs: &[]pipeline.Artifact{
+							buildArtifact,
+						},
+					}),
+				},
+			},
+			&pipeline.StageProps{
+				StageName: jsii.String("Deploy"),
+				Actions: &[]pipeline.IAction{
+					pipelineactions.NewCodeDeployEcsDeployAction(&pipelineactions.CodeDeployEcsDeployActionProps{
+						ActionName:                  jsii.String("EcsFargateDeploy"),
+						AppSpecTemplateInput:        buildArtifact,
+						TaskDefinitionTemplateInput: buildArtifact,
+						// Creates a CodeDeploy Deployment Group
+						DeploymentGroup: codedeploy.NewEcsDeploymentGroup(stack, jsii.String("CodeDeployGroup"),
+							&codedeploy.EcsDeploymentGroupProps{
+								Service: fargateService,
+								// Configurations for CodeDeploy Blue/Green deployments
+								BlueGreenDeploymentConfig: &codedeploy.EcsBlueGreenDeploymentConfig{
+									Listener:         albListener,
+									BlueTargetGroup:  targetGroupBlue,
+									GreenTargetGroup: targetGroupGreen,
+								},
+							},
+						),
+					}),
+				},
+			},
+		},
```

Comment:
NIT: break this up into three declarations, then build the array? 

This ends up embedded very deep and can get hard to read at times. I'd rely heavily on editor code folding to follow this, especially the deploy stage.